lib/swoosh/adapters/local/storage/manager.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Swoosh.Adapters.Local.Storage.Manager do
+  @moduledoc ~S"""
+  Manages the creation/monitoring of the global in-memory storage driver,
+  `Swoosh.Adapters.Local.Storage.Memory`
+  """
+
+  use GenServer
+
+  def start_link(args \\ []) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @doc false
+  def init(_args) do
+    {:ok, restart_or_monitor()}
+  end
+
+  @doc false
+  def handle_info({:DOWN, _, :process, _, :normal}, _) do
+    # In case the process is stopped normally with `stop/0`
+    {:stop, :normal, %{}}
+  end
+
+  def handle_info({:DOWN, _, :process, _, _reason}, _) do
+    # Try to either restart the global GenServer or monitor the newly
+    # created one.
```

Comment:
Is it worth adding a random duration sleep here?

Reply:
While this change already reduces the number of restarts, I think it wouldn't hurt to have a delay in case connects/disconnects are happening fast. Do you think a maximum of 2 seconds `Process.sleep(:rand.uniform(1_000) + 1_000)` is fine?

Reply:
Maybe something like 200 + rand1800 to maximize variation. What do you think?

Reply:
Sounds good, I'll add it.

Reply:
🙏